packages/library-base/src/commonMain/kotlin/io/realm/kotlin/Realm.kt

```diff
@@ -118,10 +119,17 @@ public interface Realm : TypedRealm {
             }
             if (!fileExists(configuration.path)) return false
             val config = (configuration as InternalConfiguration)
-            val (dbPointer, _) = RealmInterop.realm_open(config.createNativeConfiguration())
-            return RealmInterop.realm_compact(dbPointer).also {
-                RealmInterop.realm_close(dbPointer)
-            }
+
+            return RealmInterop.realm_create_scheduler()
+                .releaseAfter { scheduler ->
+                    val (dbPointer, _) = RealmInterop.realm_open(
+                        config = config.createNativeConfiguration(),
+                        scheduler = scheduler
+                    )
+                    RealmInterop.realm_compact(dbPointer).also {
```

Comment:
I am leaving as it it. It would be great to have a scoped-like `releaseAfter`, but the issue as you point out, is that the Realm pointer needs to be closed and then released.

It is easier to add a try-finally for this specific instance. We could consider it later if we have more occurrences for this pattern.